crates/scroll/alloy/consensus/src/transaction/tx_type.rs

```diff
@@ -0,0 +1,138 @@
+//! Contains the transaction type identifier for Scroll.
+
+use alloy_eips::eip2718::Eip2718Error;
+use alloy_primitives::{U64, U8};
+use alloy_rlp::{BufMut, Decodable, Encodable};
+use derive_more::Display;
+
+/// Identifier for an Scroll L1 message transaction
+pub const L1_MESSAGE_TX_TYPE_ID: u8 = 126; // 0x7E
+
+/// Scroll `TransactionType` flags as specified in <https://docs.scroll.io/en/technology/chain/transactions/>.
+#[repr(u8)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, Hash, Display)]
+pub enum ScrollTxType {
+    /// Legacy transaction type.
+    #[display("legacy")]
+    Legacy = 0,
+    /// EIP-2930 transaction type.
+    #[display("eip2930")]
+    Eip2930 = 1,
+    /// EIP-1559 transaction type.
+    #[display("eip1559")]
+    Eip1559 = 2,
+    /// Optimism Deposit transaction type.
+    #[display("deposit")]
+    L1Message = L1_MESSAGE_TX_TYPE_ID,
+}
+
+impl ScrollTxType {
+    /// List of all variants.
+    pub const ALL: [Self; 2] = [Self::Legacy, Self::L1Message];
+}
```

Comment:
This doesn't look like it's comprehensive.

```suggestion
impl ScrollTxType {
    /// List of all variants.
    pub const ALL: [Self; 4] = [Self::Legacy, Self::Eip1559, Self::Eip2930, Self::L1Message];
}
```